src/services/goals.js

```diff
@@ -222,22 +222,39 @@ export async function saveObjectiveAssociations(
   files = [],
   deleteUnusedAssociations = false,
 ) {
+  // We need to know if the objectiveTemplateId is populated to know if we
+  // can disable the hooks on the ObjectiveTopic
+  const o = await Objective.findOne({
+    attributes: ['objectiveTemplateId'],
+    where: { id: objective.id },
```

Comment:
Is there a possibility that objective here would be null? If so, this will throw the undefined error.  (assuming that's the desired behavior)